### PR TITLE
Do not fail on instance_info length, it is expected to be LONGTEXT

### LINKED ISSUE #48: scanTableForLongTexts also checks longtext
Hi,

the check for long texts uses `MAX_TEXT_LEN = 65536`, which is true for normal TEXT in mariadb. But not for medium or longtext(4gb), e.g. in the case of `ironic nodes.instance_info` (longtext). Here we have a false positive precheck.
Here the precheck should also take the mariadb column type into consideration.

Thanks

### PR DESCRIPTION
**Pull request by jsuchome**
With https://github.com/openstack/ironic/commit/8fdf752ae9688c581fe8c977e8a5ffb9432583f3
ironic's nodes.instance_info has changed to be LONGTEXT, not TEXT.
So we do not need to report a failure when a value in the source database
is longer then 64KiB.

This is currently specific solution to this key only, but can be expanded
to support more keys with the LONGTEXT type.

See also https://github.com/SUSE/psql2mysql/issues/48

It might be a better idea to look at the target database's schema instead of checking for the specific key. However, with current implementation, `precheck` command is supposed to work even without providing the target database access.

### CODE REVIEW

File: psql2mysql/__init__.py [hunk 1]
```diff
@@ -368,8 +380,8 @@ def do_prechecks(config, source, target):
         long_values = db.scanTableForLongTexts(table)
         if long_values:
             print("Table '%s' contains TEXT values that are more than %s "
-                  "characters long. This is incompatible with MariaDB setup.",
-                  table.name, MAX_TEXT_LEN)
+                  "characters long. This is incompatible with MariaDB setup." %
```
Comment (skazi0) [inline psql2mysql/__init__.py:383]: do you know why we mix `print()` with `LOG`?

File: psql2mysql/__init__.py [hunk 2]
```diff
@@ -98,6 +99,15 @@ def _query_long_text_rows(self, table, textColumns):
         rows = q.execute()
         return rows
 
+    # For some cases MySQL is expecting LONGTEXT instead of TEXT.
+    # We do not need to report such values as too long.
+    def longtextExpected(self, table_name, column):
+        if (self.db_path == "/ironic" and table_name == "nodes"
```
Comment (skazi0) [inline psql2mysql/__init__.py:105]: Is it possible to configure ironic with different db name?

File: psql2mysql/__init__.py (hunk 1 above)
Comment (jsuchome) [reply psql2mysql/__init__.py:383]: I think `print` was meant as the CLI output that is always shown, while log should have optional log levels and can (probably) be redirected to the log file...
That's why I think we did it this way in the first time...

File: psql2mysql/__init__.py (hunk 2 above)
Comment (jsuchome) [reply psql2mysql/__init__.py:105]: Good question. I'm quite sure we do not allow that in SOC. It's possible some other deployer does it but what would be the point?

In such case this check would obviously not catch the case. I'm not sure if it makes sense to have a special argument just for this (IMO) corner case.

Comment (skazi0) [reply psql2mysql/__init__.py:105]: Unfortunately (for this case) we allow this: https://github.com/crowbar/crowbar-openstack/blob/master/chef/data_bags/crowbar/template-ironic.json#L37

Comment (jsuchome) [reply psql2mysql/__init__.py:105]: Oh, you got me. 
However, I still think it's a corner case, or is it not? Why would people do it?

What would be the alternative solution? As I wanted this to work without providing the info about target database, it seems the only other way is the extra config option for the tool....

File: psql2mysql/__init__.py (hunk 1 above)
Comment (skazi0) [reply psql2mysql/__init__.py:383]: Ironic itself is a corner case :o)
Lets leave it as is.

**Review by skazi0 — APPROVED** [review]

**Comment by jsuchome** [comment]
Hey @rhafer , any chance you can look at this?

**Comment by rhafer** [comment]
> Hey @rhafer , any chance you can look at this?

Feel free to merge stuff here without me looking at it (you might not like my feedback anyway ;-) ). I simply didn't have the time to look at this. And it feels wrong that fixes have to wait for that.

> It might be a better idea to look at the target database's schema instead of checking for the specific key.

I agree that would be a much better aproach.

> However, with current implementation, `precheck` command is supposed to work even without providing the target database access.

It could turn the long text checks into a strong WARNING and mention that things can only be verified for real with the target database available.

**Comment by jsuchome** [comment]
@rhafer Thanks. 
I think you are right in all cases, but I'm afraid I'm really gonna ignore it ;-)

**Pull request merged**